filebeat/harvester/log.go

```diff
@@ -57,6 +57,27 @@ func (h *Harvester) Harvest(r reader.Reader) {
 	// Makes sure file is properly closed when the harvester is stopped
 	defer h.close()
 
+	// Channel to stop internal harvester routines
+	harvestDone := make(chan struct{})
+	defer close(harvestDone)
+
+	// Closes reader after timeout or when done channel is closed
+	// This routine is also responsible to properly stop the reader
+	go func() {
+		var closeTimeout <-chan time.Time
+		if h.config.CloseTimeout > 0 {
+			closeTimeout = time.After(h.config.CloseTimeout)
+		}
+
+		select {
+		case <-closeTimeout:
+			logp.Info("Closing harvester because close_timeout was reached: %s", h.state.Source)
+		case <-h.done:
```

Comment:
with harvestDone being always closed if harvester loop quits, we still need to check for h.done? Can harvester loop 'block' and not detect h.done and will h.fileReader.Close() unblock the loop?

Maybe add doc telling about the different use-cases of each channel, as they look quite redundant as is.